doc/vlan/switchport-mode-support/Switchport Mode and VLAN CLI Enhancement.md

```diff
@@ -0,0 +1,388 @@
+
+## Definitions/Abbreviations
+
+| **Sr No** |  **Term**   |  **Definition**                      |
+| :------------- |:-------------| :-----|
+|  1       |  VLAN_ID      |  Unique identifier for each VLAN     |  
+|  2       |  Dot1q        |  IEEE 802.1Q Protocol                |  
+|  3       |  Trunk  |  The port is untagged member to one VLAN and tagged member to one or more VLANs  |  
+|  4       | Access |  The port is untagged member to only one VLAN|  
+|  5       |  Routed | The port is in L3 interface mode                       |  
+|  6       |  Hybrid|   The port is multiple untagged and multiple tagged vlan members to  more VLANs         |  
+
+
+## Overview
+
+This HLD will provide implementation details of switch port modes access and trunk on a Port or on a PortChannel. In this HLD vlan configuration commands are also improved for efficient configuration and management of VLANs.
+
+## Introduction
+
+Switch ports are Layer 2-only interfaces associated with a physical port. A switch port can have three modes: access port, a trunk port, or a hybrid port. We can manually configure a port as an access port, trunk port or hybrid port depending on requirement . Switch ports are used for managing the physical interface and associated Layer 2 protocols and do not handle routing.
+
+### Access Mode
+
+In access mode a port possesses the following properties
+* Receives untagged/tagged traffic for a particular Vlan
+* Forwards untagged/tagged traffic for a particular Vlan
+
+### Trunk Mode
+
+In trunk mode a port possesses the following properties
+* Receives untagged traffic for a single Vlan
+* Forwards untagged traffic for a single Vlan
+* Receives tagged traffic for one or more Vlan(s)
+* Forwards tagged traffic for one or more Vlan(s)
+
+### Hybrid Mode
+
+In hybrid mode a port possesses the following properties
+* Receives  multiple tagged & untagged traffic for  one or more Vlan(s)
+* Forwards multiple tagged & untagged traffic for  one or more Vlan(s)
+
+### Sample Topology
+
+![Topology](https://user-images.githubusercontent.com/61490193/208155234-3a04ca08-5ca3-42b3-b679-c1efcacddff2.png)
+
+
+__*Figure 1: Sample Topology*__
+
+The figure shows that 3 different VLANs are configured with 6 other hosts. The switch ports connected to hosts would be configured as access ports. The ports that connect switches together would be configured as trunk ports.
+
+## Architecture Design
+
+The overall SONiC architecture will remain the same and no new sub-modules will be introduced. Changes are made only in the CLI container and Config_DB.
+
+![Architecture](https://user-images.githubusercontent.com/61490193/217232727-811b98a8-ce9c-4227-a94a-1c83a4825a62.png)
+
+
+## High-level Design
+
+In this section we will explain sequence diagrams for the implemented features.
+
+### Switch Port Mode
+
+![SwitchPortMode-PortPortChannel](https://user-images.githubusercontent.com/61490193/207826139-3ff0c41a-69af-4018-bf19-c083e1307bf7.png)
+
```

Comment:
This sequence diagram doesn't clearly show the separation between the processing done in the CLI vs that done in the clicommon. For example, the same "if" checks are done in both the components.

Reply:
Added state transition diagram for Switchport modes in HLD for more clarity. Thanks